src/scripts/BackgroundScript.js

```diff
@@ -38,17 +36,18 @@ let BackgroundScript = {
 	 * @param request.fn	function to call
 	 * @param request.request	request that will be passed to the function
 	 */
-	executeContentScript: function(request) {
+	executeContentScript: function (request) {
 		var reqToContentScript = {
 			contentScriptCall: true,
 			fn: request.fn,
 			request: request.request,
 		};
 		var deferredResponse = $.Deferred();
-		this.getActiveTabId().then(function(tabId) {
-			browser.tabs.sendMessage(tabId, reqToContentScript).then(function(response) {
+		this.getActiveTabId().then(function (tabId) {
+			browser.tabs.sendMessage(tabId, reqToContentScript).then(function (response) {
 				deferredResponse.resolve(response);
 			});
+			deferredResponse.reject(err => reject(err));
 		});
```

Comment:
Also suggest to add .catch() and deferredResponse.reject(...) after getActiveTabId promise